internal/webhook/external/github.go

```diff
@@ -190,6 +192,45 @@ func (g *githubWebhookReceiver) getHandler(requestBody []byte) http.HandlerFunc
 			// that URL, we will miss any Warehouses that are subscribed to the same
 			// repository using a different URL format.
 			repoURLs = []string{git.NormalizeURL(e.GetRepo().GetCloneURL())}
+
+		case *gh.RegistryPackageEvent:
+			switch e.GetAction() {
+			// These are the only actions that should refresh Warehouses.
+			case "published", "updated":
+			default:
+				xhttp.WriteResponseJSON(w, http.StatusOK, nil)
+				return
+			}
+			pkg := e.GetRegistryPackage()
+			if pkg == nil || pkg.GetPackageVersion() == nil {
+				xhttp.WriteErrorJSON(
+					w,
+					xhttp.Error(errors.New("invalid request body"), http.StatusBadRequest),
+				)
+				return
+			}
+			switch pkg.GetPackageType() {
+			// These are the only types of packages we care about.
+			case ghcrPackageTypeContainer, ghcrPackageTypeDocker:
+			default:
+				xhttp.WriteResponseJSON(w, http.StatusOK, nil)
+				return
+			}
+			pkgURL := pkg.GetPackageVersion().GetPackageURL()
+			// GitHub sometimes sends package URLs with a trailing colon
+			// and no tag (e.g., "ghcr.io/user/image:"). Such strings are
+			// not valid OCI image references and will result in empty
+			// repo URLs. We trim the trailing colon to avoid parsing errors
+			// and optionally allow fallback handling (e.g., ":latest").
+			if strings.HasSuffix(pkgURL, ":") {
+				pkgURL += "latest"
+			}
```

Comment:
Why are you putting a latest tag on the end of this?

We want a URL for the _repository_. You are incorrectly are constructing a URL for a specific image _within_ the repository -- one that may not even exist.

And when normalized, the tag will be stripped away anyway.

You're manipulating the URL in a totally nonsensical way and then it's almost immediately undone by existing logic.

Stripping the trailing colon, if it exists is the only thing you should be doing here.

```suggestion
			pkgURL = strings.TrimSuffix(pkgURL, ":")
```

Reply:
resolved.

Reply:
okay so we can't officially call them as `package_url` even though GitHub writes it as `package_url`, correct?